tests/test_cli.py

```diff
@@ -176,10 +185,176 @@ def test_correct_call_DownloadManager_with_relative_time(
+
+    def test_extract_arg(self):
+        """shall return the correct argument values based on the 'sys.argv' """
+        sys.argv = [
+            "",
+            '-d', "/test_folder/",
+            '-b', "200",
+            '-e', "2018-11-18",
+            '-r',
+            '-f',
+            '-l',
+            '-g', "POLYGON ((-22 84, -22 74, 32 74, 32 84, -22 84))",
+            '-t', 'test_type',
+            '-q',
+            '{"dataseturi__uri__contains": "osisaf", "source__instrument__short_name__icontains": '
+            + '"AMSR2"}'
+        ]
+        arg = cli_copy.cli_parse_args()
+        self.assertEqual(arg.begin, '200')
+        self.assertEqual(arg.destination_path, '/test_folder/')
+        self.assertEqual(arg.end, '2018-11-18')
+        self.assertEqual(arg.geometry, 'POLYGON ((-22 84, -22 74, 32 74, 32 84, -22 84))')
+        self.assertEqual(arg.type, 'test_type')
+        self.assertEqual(arg.query,
+                         '{"dataseturi__uri__contains": "osisaf", '
+                         + '"source__instrument__short_name__icontains": "AMSR2"}')
+        # testing the flag presence
+        self.assertTrue(arg.rel_time_flag)
+        self.assertTrue(arg.flag_file)
+        self.assertTrue(arg.link)
+        sys.argv.remove('-r')
+        sys.argv.remove('-f')
+        sys.argv.remove('-l')
+        arg = cli_copy.cli_parse_args()
+        self.assertFalse(arg.rel_time_flag)
+        self.assertFalse(arg.flag_file)
+        self.assertFalse(arg.link)
+
+    def test_lack_of_calling_json_deserializer_when_no_query_appears_for_copying(self):
+        """'json.loads' should not called when nothing comes after '-q' """
+        sys.argv = [
+            "",
+            '-d', "/test_folder/",
+            '-b', "200",
+            '-e', "2018-11-18",
+            '-r',
+            '-f',
+            '-l',
+            '-g', "POLYGON ((-22 84, -22 74, 32 74, 32 84, -22 84))",
+            '-t', 'test_type'
+        ]
+        with mock.patch('json.loads') as mock_json:
+            cli_copy.main()
+        mock_json.assert_not_called()
+
+    @mock.patch('os.path.isfile', side_effect=[True, False, True, False])
+    # The even side effects (the 'True' ones) are associated to the destination and the odd ones are
+    # associated to the source path. It is because 'os.path.isfile' is used for evaluating both
+    # source paths and destination paths.
+    def test_correct_destination_folder_for_all_files_that_are_copied(self, mock_isfile):
+        """ the copied file(s) shall be copied at the destination folder """
+        sys.argv = [
+            "",
+            '-b', "2018-04-01",
+            '-e', "2018-04-09",
+            '-d', "/dst_folder/"
+        ]
+        for dataset in Dataset.objects.all():
+            dataset.dataseturi_set.get_or_create(
+                uri='/new_loc_add', dataset=dataset, service=LOCAL_FILE_SERVICE)
```

Comment:
That should be in fixtures.

Reply:
Fixtures are designed carefully for having all possible situations in them, nothing more and nothing less. Here we have 3 datasets in the fixture which have below list of dataset_uri (numbers of datasets maybe not correct in the list below):
3. healthy url
1.healthy url **_plus_** fake url
2.healthy url **_plus_** the second fake url **_plus_** local address

In these three comment line I add another local address named `/new_loc_add` for **all** dataset! This means if we move it to fixtures, the fixtures will be polluted and we have to change a lot of tests accordingly.

The reason why I add that these lines is that maybe someone download ALL the datasets once again in another folder based on his/her choice, so in that case, copy should be able to access and search both of addresses stored in the database. 

So it is better to have this code here because they are only belong to this test! only!

I will change the docstring of the test based on this comment of yours.

Reply:
In these two tests you need some data for testing. The standard way of providing data in the database for testing is using fixtures.
Please move to fixtures.